compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -54,6 +38,16 @@ class InterceptedMethods extends MiniPhase {
     else tree
   }
 
+  override def transformIdent(tree: tpd.Ident)(implicit ctx: Context): Tree = {
```

Comment:
Can you factor out the common code between `transformIdent` and `transformSelect`

Reply:
What about something like:
```scala
private def transformRefTree(tree: RefTree)(implicit ctx: Context): Tree =
  if (tree.symbol.isTerm && (defn.Any_## eq tree.symbol.asTerm)) {
    val qual = tree match
      case id: Ident =>tpd.desugarIdentPrefix(id)
      case sel: Select => sel.qual
    }
    val rewrite = poundPoundValue(qual)
    ctx.log(s"$phaseName rewrote $tree to $rewrite")
    rewrite
  }
  else tree

override def transformIdent(tree: tpd.Ident)(implicit ctx: Context) =
  transformPoundPound(tree)

override def transformSelect(tree: tpd.Select)(implicit ctx: Context): Tree =
  transformPoundPound(tree)
```

Reply:
Done

Reply:
Though it added a line of code